website/content/docs/concepts/integrated-storage.mdx

```diff
@@ -169,15 +179,117 @@ node2    node2.vault.local:8201    follower    true
 node3    node3.vault.local:8201    leader      true
 ```
 
-## Server-to-Server Communication
+## Integrated Storage and TLS
 
-Once nodes are joined to one another they begin to communicate using mTLS over
-Vault's cluster port. The cluster port defaults to `8201`. The TLS information
-is exchanged at join time and is rotated on a cadence.
+We've glossed over some details in the above sections on bootstrapping clusters.
+The instructions are sufficient for most cases, but some users have run into
+problems when using auto-join and TLS in conjunction with things like auto-scaling.
+The issue is that go-discover on most platforms returns IPs (not hostnames), and
+because the IPs aren't knowable in advance, the TLS certificates used to secure
+the Vault API port don't contain these IPs in their IP SANs.
 
-A requirement for integrated storage is that the
-[cluster_addr](/docs/concepts/ha#per-node-cluster-address) configuration option
-is set. This allows Vault to assign an address to the node ID at join time.
+### Vault networking recap
+
+Before we explore solutions to this problem, let's recapitulate how Vault nodes
+speak to one another.
+
+Vault exposes two TCP ports: the API port and the cluster port.
+
+The API port is where clients send their Vault HTTP requests.
+
+For a single-node Vault cluster you don't worry about a cluster port as it won't be used.
+
+When you have multiple nodes you also need a cluster port.  This is used by Vault
+nodes to issue RPCs to one another, e.g. to forward requests from a standby node
+to the active node, or when Raft is in use, to handle leader election and
+replication of stored data.
+
+The cluster port is secured using a TLS certificate that the Vault active node
+generates at startup.  It's clear how this can work when not using integrated
```

Comment:
The TLS information used for the raft server has a different lifecycle than the request forwarding server. Raft TLS isn't refreshed on startup, it's generated on vault init and then rotated on a cadence. To generalize a bit more:

```suggestion
generates internally.  It's clear how this can work when not using integrated
```

Reply:
Done.